flipperdevices/flipperzero-firmware
# TLSF memory allocator. Less free flash, moar free ram.

# What's new

- Heap4 allocator is replaced by tlsf.
- Moar free heap, less free flash.
- Updater: properly read large update file.

# Verification 

- Run `unit_tests`.
- In general, everything is affected.
- Update from any previous version to this one compiled with `LIB_DEBUG=1` will not work (updater bug, see [cab517f](https://github.com/flipperdevices/flipperzero-firmware/pull/3572/commits/cab517fb2647d65107dc9bd9dd0872519847dee7))
- Update from this version to this one compiled with `LIB_DEBUG=1` will work

# Checklist (For Reviewer)
